integration-tests/src/utils/node_proc.rs

```diff
@@ -168,22 +168,30 @@ fn find_substrate_port_from_output(r: impl Read + Send + 'static) -> u16 {
     BufReader::new(r)
         .lines()
         .find_map(|line| {
-            let line = line
-                .expect("failed to obtain next line from stdout for port discovery");
+            let line =
+                line.expect("failed to obtain next line from stdout for port discovery");
 
             // does the line contain our port (we expect this specific output from substrate).
-            let line_end = match line.rsplit_once("Listening for new connections on 127.0.0.1:") {
-                None => return None,
-                Some((_, after)) => after
+            let line_end = match line
+                .rsplit_once("Listening for new connections on 127.0.0.1:")
+            {
+                None => {
+                    match line.rsplit_once("Running JSON-RPC WS server: addr=127.0.0.1:")
```

Comment:
you could also do: https://github.com/paritytech/substrate/blob/master/bin/node/cli/tests/common.rs#L164-#L167

then just parse is as `SocketAddr` then call port on it.

Reply:
sorry, I didn't think about making the log easy to parse :)

Reply:
For the testing pov I guess it's ok assuming 127.0.0.1 though (ie you could start it on 0.0.0.0, but you couldn't then connect to 0.0.0.0)?

Reply:
it's fair assumption, I was thinking that some weird platforms could actually have some other loopback address than 127.0.0.1. Picking hair here just ignore me but I was more thinking about having to avoid parsing the port manually :P

I think you can connect to 0.0.0.0 but it means all addresses IIRC.